app/models/piece.rb

```diff
@@ -2,4 +2,80 @@ class Piece < ActiveRecord::Base
 
   belongs_to :game
   belongs_to :user, class_name: 'User'
+
+  def obstructed?(destination_row, destination_col)
+
+=begin  * Sample Chessboard *
+            0 1 2 3 4 5 6 7
+            - - - - - - - -
+        0 | 0 1 0 0 0 1 0 0
+        1 | 0 0 0 0 0 0 0 0
+        2 | 0 0 0 0 0 0 0 0
+        3 | 0 0 0 0 0 0 0 0
+        4 | 0 0 0 0 0 0 0 0
+        5 | 0 0 0 0 0 0 0 0
+        6 | 0 0 0 0 0 0 0 0
+        7 | 0 0 0 0 0 0 0 0
+=end
+
+    # determine if we should march forward of backwards
+    # actually march forwards and backwards to build an array of spaces
+    # determine if a piece is in the square from the position.
+
+    current_piece_column_index = self.current_column_index
+    current_piece_row_index = self.current_row_index
+    # Determine number of spaces we need to check
+      row_diff = destination_row - current_piece_row_index
+      col_diff = destination_col - current_piece_column_index
+
+      delta = something ? 1 : -1
+
+    #HORIZONTAL CHECK
+    # spaces = []
+
+    # current_x_position = self.current_column_index - delta
+    # while current_x_position != self.current_column_index - delta do
+    #   spaces << [current_x_position, destination_y]
+    #   current_x_position += delta
+    # end
+
+    # while col_diff != current_piece_column_index do
+    #   spaces.push(curent_piece_row_index,col_diff)
+    #   col_diff -= 1
+    # end
+
+    # if current_pos = 1,1, destination is 5,1, then array is
+    # spaces= [5,1], [4,1], [3,1], [2,1]
+    # spaces.each do |row_index, col_index|
+
+    # if game.pieces(row_index, col_index) != nil
+    #    return true
+    #  end
+  # end
+
+  end
```

Comment:
You're close!

This is what I recommend. Looks like you've already noticed that it's really 3 problems. `vertical`, `horizontal` and `diagonal`.

Try solving one at a time.

**First**
You'll need methods that determine what kind of move it is.

example:
```Ruby
def horizontal?(destination_row, destination_col)
  # is the row the same, but the col different?
  (current_row_index == destination_row) && (current_col_index != destination_col)
end
```

I'm thinking it could be used like this

```Ruby
def obstructed?(destination_row, destination_col)
  return horizontal_move(destination_row, destination_col) if horizontal?(destination_row, destination_col)
  return vertical_move(destination_row, destination_col) if vertical?(destination_row, destination_col)
  # ... ect ...
end
```


**Next**
Now that you can determine the type of move, you can use your `horizontal_move` method to check for obstructions.

I recommending using a loop. Something like this would check all pieces in a row.

Hint: use range generation to get the in between positions. http://www.tutorialspoint.com/ruby/ruby_ranges.htm

```Ruby
(current_row_index + 1...destination_row).each do |row|
  return true if game.pieces.where(current_row_index: row, current_col_index: col).exists?
end
```

Can use `exists?` to see if a piece exists in the space you're checking :smile: 

Let me know if you have questions.